JobConfig/pions/PionsStage1_2020.fcl

```diff
@@ -0,0 +1,146 @@
+// Andrei Gaponenko, 2013
+// Jake Colston & Joseph Brock, 2015 (edit for cd3)
+
+#include "fcl/minimalMessageService.fcl"
+#include "fcl/standardProducers.fcl"
+#include "fcl/standardServices.fcl"
+
+BEGIN_PROLOG
+
+// Energy cuts on electron, positron, photon, and neutron.
+CutEPPhN : {
+    type : intersection
+    pars : [ { type: kineticEnergy cut: 100.0 },
+             { type: pdgId pars: [ 11, -11, 22, 2112 ] } ]
+}
+
+END_PROLOG
+
+# Give this job a name.
+process_name : pionsg4s1
+
+source : {
+    module_type : EmptyEvent
+    //maxEvents   : 1000000
+}
+
+services : {
+    message                : @local::default_message
+    TFileService           : { fileName : "nts.owner.pions-g4s1.version.sequencer.root" }
+    RandomNumberGenerator  : {defaultEngineKind: "MixMaxRng"}
+    GeometryService        : { inputFile      : "JobConfig/common/geom_baseline.txt"    }
+   
+    ConditionsService      : { conditionsfile : "Mu2eG4/test/conditions_01.txt"      }
+    GlobalConstantsService : { inputFile      : "Mu2eG4/test/globalConstants_01.txt" }
+    G4Helper               : {}
+    SeedService            : @local::automaticSeeds
+}
+
```

Comment:
Please use the official definitions of the standard services for Sim or SimReco as appropriate; then override anything that you need to.  If you don't do this then, when the standard changes you need to edit every file that you just made.

Reply:
Look, for example at: JobConfig/mixing/NoPrimary.fcl

The services should be defined like:
services: @local::Services.Sim

with overrides for the TFileService file and geometry file if necessary.

The configuration of g4 should be:

producers: {
  g4run : @local::g4run
}

with overrides as needed.  Note that the module label "g4un" has an important meaning since downstream code uses it to find their data products.

For most of the other modules you use there are predefined standard definitions.  Search the prolog.fcl file to find them.  



Also look at JobConfig/mixing/NoPrimaryPBISequence.fcl   . Note that it says to do the same job as the NoPrimary.fcl one but with a small list of changes.  

This reason for the pattern of deep includes is that it is the only solution to minimizing multiple points of maintenance when a fcl parameter set need so change.  The disadvantage of deep includes is that it makes it more difficult to understand what the fcl file is doing; we have tools to help with this.  Do you know how to use:

 mu2e --debug-config --annotate
 mu2e --config-summary 

JobConfig has files with a mix of vintages.   The older files still have practices that we have long improved on.  We have been updating them as we need to use them again - the cost in people time of systematically changing everything, and testing it, is just too high.